main.go

```diff
@@ -668,7 +682,24 @@ func clientTransport(ctx *cli.Context, enableTLS bool) http.RoundTripper {
 			ClientSessionCache:       tls.NewLRUClientSessionCache(tlsClientSessionCacheSize),
 		}
 	}
+	rtw.transport = tr
+}
 
+func (rtw *roundTripperWrapper) RoundTrip(req *http.Request) (*http.Response, error) {
+	rtw.rwLocker.RLock()
+	defer rtw.rwLocker.RUnlock()
+	return rtw.transport.RoundTrip(req)
+}
```

Comment:
Can't we make `rtw.transport` an atomic pointer?

It seems like a bad idea that this will block until 1) All other requests have finished and 2) The transport has been replaced.